src/app/modules/auth/auth.controller.ts

```diff
@@ -217,3 +227,92 @@ export const handleSignout: RequestHandler = async (req, res) => {
     return res.status(StatusCodes.OK).json({ message: 'Sign out successful' })
   })
 }
+
+/**
+ * NOTE: This is exported only for testing
+ * Generates redirect URL to Official SingPass/CorpPass log in page
+ * @param isPersistentLogin whether the client wants to have their login information stored
+ * @returns 200 with the redirect url when the user authenticates successfully
+ * @returns 400 when there is an error on the authType of the form
+ * @returns 400 when the eServiceId of the form does not exist
+ * @returns 404 when form with given ID does not exist
+ * @returns 500 when database error occurs
+ * @returns 500 when the redirect url could not be created
+ * @returns 500 when the redirect feature is not enabled
+ */
+export const getRedirectLink: RequestHandler<
```

Comment:
I think this function should be in the public-form controller instead? The auth controller is for admin login and logout

Reply:
ah ok. good point! i'll shift this over to the appropriate folder!